pkg/api/types.go

```diff
@@ -396,6 +398,24 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts are Glusterfs hosts address array
+	Hosts []string `json:"hosts"`
+
+	// Required: Path is the Glusterfs volume path
+	Path string `json:"path"`
+
+	// Optional: mountOptions is the mount time options
+	// including readonly
+	MountOpt string `json:"mountOptions,omitempty"`
```

Comment:
Yes, there are a few mount time options (mostly from FUSE) that have security and performance impact. 

----- Original Message -----
From: "Eric Tune" <notifications@github.com>
To: "GoogleCloudPlatform/kubernetes" <kubernetes@noreply.github.com>
Cc: "Huamin Chen" <hchen@redhat.com>
Sent: Tuesday, March 31, 2015 11:55:36 AM
Subject: Re: [kubernetes] implement glusterfs volume plugin (#6174)

> @@ -396,6 +398,24 @@ type NFSVolumeSource struct {
>  	ReadOnly bool `json:"readOnly,omitempty"`
>  }
>  
> +// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
> +type GlusterfsVolumeSource struct {
> +	// Required: hosts are Glusterfs hosts address array
> +	Hosts []string `json:"hosts"`
> +
> +	// Required: Path is the Glusterfs volume path
> +	Path string `json:"path"`
> +
> +	// Optional: mountOptions is the mount time options
> +	// including readonly
> +	MountOpt string `json:"mountOptions,omitempty"`

Are there other mount options that are useful with glusterfs?

---
Reply to this email directly or view it on GitHub:
https://github.com/GoogleCloudPlatform/kubernetes/pull/6174/files#r27492715